### DISCUSSION TITLE
When Kong proxy TLS service, SNI fails to transparently transmit the problem

### DISCUSSION THREAD

**Discussion by Murphy-hub** [Kong/kong#8190] [Help] [answered]
When I use Kong as the proxy, the TLS protocol is used in route and the TLS protocol is also used in service. An exception occurs. The SNI obtained by my actual service is Kong_ Upstram, instead of the SNI configured in my route, can't transmit SNI through. I'm a little confused. How should I operate
![image](https://user-images.githubusercontent.com/52234994/145948851-bfd513d3-f72f-41e6-9ec2-67d9a5783fad.png)
![image](https://user-images.githubusercontent.com/52234994/145948944-b78fd8af-28d7-4c9c-b7ea-1d3736034e4a.png)

**Comment by kikito** [comment]
Hello,

What version of Kong are you using? Any plugins? Could you try with direct JSON requests to Kong's Admin API instead of using a graphical interface (I assume you are using Konga)?

>  An exception occurs

Do you get a response when that happens? What is the response? Also, what do your logs say?

**Comment by mayocream** [comment, marked as answer]
Hi @zhangshuaiNB, Thanks for reporting this.

It seems like your PR [Support TLS SNI custom dynamic configuration #8196](https://github.com/Kong/kong/pull/8196) is a workaround in this case, and we will continue to discuss passing the original SNI name to the upstream service in Kong by default.

You can also try [Proxy TLS passthrough traffic](https://docs.konghq.com/gateway/2.7.x/reference/proxy/#proxy-tls-passthrough-traffic) feature in Kong 2.7.0.
